src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -345,7 +346,30 @@ private HttpMessageHandler SetupHandlerChain()
                 return Task.FromException<HttpResponseMessage>(error);
             }
 
-            return handler.SendAsync(request, cancellationToken);
+            return HttpTelemetry.IsEnabled && request.RequestUri != null ?
+                WithLogging(handler, request, cancellationToken) :
+                handler.SendAsync(request, cancellationToken);
+
+            static async Task<HttpResponseMessage> WithLogging(HttpMessageHandler handler, HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                Debug.Assert(request.RequestUri != null);
+                HttpTelemetry.Log.RequestStart(
+                    request.RequestUri.IdnHost,
+                    request.RequestUri.Port,
+                    request.RequestUri.Scheme,
+                    request.RequestUri.AbsolutePath,
+                    request.Version.ToString());
+                try
+                {
+                    return await handler.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                }
+                catch
```

Comment:
Debugging exception noise is lessened if you use an exception filter rather than catch/rethrow

Reply:
@noahfalk What "noise" are you referring to? In general, we want to catch everything here so that we can guarantee generating the 'RequestAborted' event.

Reply:
A try/catch/throw pattern throws the exception twice, so the debugger will potentially stop on it twice with a UI popup and log it twice. A little more annoyingly if the debugger is configured only to stop on unhandled exceptions (the default setting) then at best the debugger will stop where the exception is rethrown here but the initial location of the throw will never be considered unhandled because of this catch clause. This in turn means stack frames and useful local variables at the original throw site have already been unwound when the debugger stops later. Async code has similar catch/rethrow behavior so chances are user experience still won't be ideal, but I'd be more optimistic that the debugger can find ways to mitigate the issues caused by async in the future whereas its harder to mitigate explicitly written try/catch/throw clauses with arbitrary logic inside.

Doing a filter shouldn't interfere with you being able to detect all thrown exceptions. If you haven't seen it before the pattern looks like this:
````
try
{
    Console.WriteLine("throw something");
    throw new AnyException();
}
catch (Exception e) when HandleException(e)
{
    Debug.Fail() // this never runs
}

bool HandleException(Exception e)
{
    Console.WriteLine("Exception occured: " + e.GetType());
    return false; // returning false means the catch handler isn't run, the exception isn't considered to be caught so there is no need to throw it again
}
````

Reply:
Thanks! This is a good pattern. I'll incorporate this suggestion.